libs/platform/src/lib/components/form/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,295 @@
+    EventEmitter,
+    Input,
+    NgZone,
+    Optional,
+    Output,
+    Self,
+    ViewChild,
+    ViewEncapsulation
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import { CheckboxComponent as CoreCheckboxComponent } from '@fundamental-ngx/core';
+import { BaseInput } from '../base.input';
+
+/**
+ * Checkbox implementation based on the
+ * https://github.com/SAP/fundamental-ngx/wiki/Platform:-Checkbox-Component-Technical-Design
+ * documents.
+ *
+ *
+ */
+@Component({
+    selector: 'fdp-checkbox',
+    templateUrl: './checkbox.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None
+})
+export class CheckboxComponent extends BaseInput implements AfterViewInit {
+    /** set to true if binary checkbox */
+    @Input()
+    isBinary: boolean = false;
+
+    /** Sets label for checkbox. */
+    @Input()
+    label: string;
+
+    /**
+     * Attached to the aria-label attribute of the host element. In most cases, aria-labelledby will
+     * take precedence so this may be omitted.
+     */
+    @Input('aria-label')
+    ariaLabel: string = '';
+
+    /**
+     * Users can specify the `aria-labelledby` attribute which will be forwarded to the input element
+     */
+    @Input('aria-labelledby')
+    ariaLabelledby: string | null = null;
+
+    /** indeterminate state */
+    @Input()
+    tristate: boolean = false;
+
+    /** indeterminate state can be selected  */
+    @Input()
+    tristateSelectable: boolean = true;
+
+    /** value for checkbox, when it is not binary */
+    @Input()
+    get value(): any {
+        return this._checkboxValue;
+    }
+    set value(selectValue: any) {
+        if (selectValue !== this.value) {
+            this._checkboxValue = selectValue;
+        }
+    }
+
+    /**
+     * Whether the checkbox is checked.
+     */
+    @Input()
+    get checked(): boolean {
+        return this._checked;
+    }
+    set checked(value: boolean) {
+        if (value !== this.checked) {
+            this._checked = value;
+        }
+    }
+
+    @Output()
+    readonly checkedChange = new EventEmitter();
+
+    /** Emitting checkbox change event */
+    @Output()
+    readonly change = new EventEmitter();
+
+    /** Event emitted when the checkbox's `indeterminate` value changes. */
+    @Output()
+    readonly indeterminateChange = new EventEmitter();
+
+    /** @hidden
+     * tracking checkbox current value
+     */
+    public checkboxCurrentValue: any;
+
+    /* @hidden
+     * reference of child checkbox implementation
+     */
+    @ViewChild(CoreCheckboxComponent)
+    private corecheckbox: CoreCheckboxComponent;
+
+    /* @hidden
+     * stores  formControl values
+     */
+    private multiSelectModel: any;
+
+    /**
+     * @hidden checkbox state, used when checkbox is used without form.
+     */
+    private _checked: boolean = false;
+
+    /** @hidden value of checkbox */
+    private _checkboxValue: any;
```

Comment:
we have discussed also this extra property to save the value. do we need it? Why we cannot directly save this to value we have already there. 

To save the selection we have this multiselectModel

Reply:
solution:
	option 1: saving control value to this._value(in baseInput) by using super.writeValue(value); this._value is different from checkbox value(value when checked), so i have used an extra variable.

Reply:
have used a different variable to store checkbox value.
setValue() and getValue() is used and sets control value on checkbox change and while writeValue().